pkg/operator/events/recorder_deployment.go

```diff
@@ -0,0 +1,35 @@
+package events
+
+import (
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsv1client "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/tools/reference"
+)
+
+// eventSourceReplicaSetNameEnv is a name of environment variable inside container that specifies the name of the current replica set.
+// This replica set name is then used as a source/involved object for operator events.
+const eventSourceReplicaSetNameEnv = "EVENT_SOURCE_REPLICASET_NAME"
+
+// eventSourceReplicaSetNameEnvFunc allows to override the way we get the environment variable value (for unit tests).
+var eventSourceReplicaSetNameEnvFunc = func() string {
+	return os.Getenv(eventSourceReplicaSetNameEnv)
+}
+
+// DeploymentFromEnvRecorderSource return event recorder source using the EVENT_SOURCE_REPLICASET_NAME environment variable
+// set inside the container created by deployment.
+// TODO: Switching to a lister will make this faster.
```

Comment:
it's called once per process, isn't  it?

Reply:
yeah, this todo is not relevant